Line Ending Conversion Corrections

bwmilby opened the pull request:
In two cases, native Mac line endings are not being converted (CRLF and LFCR are converted to LF but CR was not touched).
In one case, the constant CR was used where it should have been numToCodepoint(13).
Changed to use the LF constant to make it more readable.
Also changed numToChar to numToCodepoint since the former function is deprecated.

bwmilby committed:
Line Ending Conversion Corrections

In two cases, native Mac line endings are not being converted (CRLF and LFCR are converted to LF but CR was not touched).
In one case, the constant CR was used where it should have been numToCodepoint(13).
Changed to use the LF constant to make it more readable.
Also changed numToChar to numToCodepoint since the former function is deprecated.

bwmilby commented:
In other work I've done, LFCR has been considered invalid and removed from line ending conversion code.  I did not do that here, but could do so.  This would also apply to https://github.com/livecode/livecode-ide/pull/1853

runrevmark commented:
@bwmilby I remember discussing that with you - which patch was it we eliminated LFCR? I think the general pattern we should use for this through out is:

```
replace CR & LF with LF
replace CR with LF
```

As this is slightly simpler and codifies the line-endings which LiveCode understands as LF, CR and CRLF.

bwmilby commented:
This is where we previously discussed the issue of LFCR: https://github.com/livecode/livecode/pull/5792

I believe the pattern would need to be (LCS):
```
replace CRLF with LF in ...
replace numToCodepoint(13) with LF in ...
```

runrevmark commented:
@bwmilby : You are correct, of course - I should have made my code a little less LCS-like to avoid confusion :)

bwmilby commented:
I guess that is the “problem” with an English-like programming language :)

bwmilby committed:
Remove LFCR replacements

docs_builder.livecodescript
	private function escape pString, pConvertLineEndings

revdocsparser.livecodescript
	private function escape pString, pConvertLineEndings

bwmilby committed:
Merge branch 'develop-8.2' into bwm-CRfix

bwmilby committed:
Merge branch 'develop-8.2' into bwm-CRfix

bwmilby commented:
There is a similar issue in ide-support/revhtml5urllibrary.livecodescript ~line 441.  Will need to submit a PR against develop for that one.

montegoulding commented on ide-support/revdocsparser.livecodescript:
```diff
@@ -2169,10 +2169,10 @@ private function escape pString, pConvertLineEndings
+      replace numToCodepoint(13) with "\n" in pString
    end if
-   replace CR with "\n" in pString
+   replace LF with "\n" in pString
```
@bwmilby same issue as I noted on the IDE patch there's a problem with `pConvertLineEndings` false here potentially not escaping carriage return

bwmilby replied:
@montegoulding all uses in this file omit the second parameter which means that it will be false (27).  Many times it is an actual literal like "doc" or a variable that holds an array key.
since text is converted upon import to LF and we are not seeing bugs in the docs related to this, I don't think it is an issue.
Having said that, adding an else would make it "complete" in case a carriage return did make it in did arise.

bwmilby committed:
Line Ending Conversion Corrections

Add handling for the case that an orphan carriage return character is found and for when not converting line endings.

montegoulding commented:
@livecode-vulcan review ok f8c5d0d

livecode-vulcan commented:
💙 review by @montegoulding ok f8c5d0d5ab8df30f15843beb771110435ba2a56a

livecode-vulcan commented:
😎 test success f8c5d0d5ab8df30f15843beb771110435ba2a56a

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1126)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1142)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1116)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/69)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/69)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1121)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1143)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1126)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1142)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1125)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1135)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1159)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1121)
